### ISSUE TITLE
proper file change watching

### ISSUE THREAD

**Issue by tarpas** [tarpas/pytest-testmon#4]
the tmon.py is a primitive script put together for demo purposes. I would be very glad to put the re-running out of scope ot testmon. 
to consider:

* watchdog has https://github.com/gorakhargosh/watchdog#shell-utilities 
* xdist has http://pytest.org/latest/xdist.html#running-tests-in-looponfailing-mode (which maybe should be out of py.test or it's plugins?
* there is https://github.com/guard/guard but it's ruby

In short term, testmon will probably be failing with it's bugs so I will not want to run it automatically that much. Also in long term I quite struggle with console test outputs so I would probably run py.test manually or use PyCharm to run a test suite.

**Comment by blueyed** [comment]
Yes, any external tool like `watchmedo` from watchdog would do.

A few examples might be provided in the documentation, and/or as example files - keeping the tmon.py script, but e.g. not installed automatically - or with a different name.

I think that testmon might also provide a default "tricks" file for watchdog maybe.

**Comment by blueyed** [comment]
There is also https://github.com/seb-m/pyinotify and https://github.com/copton/react/ using it (the latter (currently) leaks however, see PRs).

**Comment by blueyed** [comment]
It might be nice if testmon would not need to restart py.test at all, but could loop over it internally, using e.g. watchdog.
That would certainly improve performance / the feedback loop, since py.test takes some time during the collection phase alone.

Just some food for thought.

**Comment by tarpas** [comment]
To looping over internally: The biggest cost of restart at the moment seems to be test collection. And we do want to have freshly created tests collected so the real task here is optimizing pytest's collection (e.g. using cache)

And another contender for the functionality
pytest --looponchange
https://bitbucket.org/pytest-dev/pytest/pull-request/253/take-in-pytest-cache/
As far as I could understand it uses polling :(

**Comment by tarpas** [comment]
Polytester seems to be a very nice tool for this: https://github.com/skoczen/polytester

**Comment by tarpas** [comment]
https://github.com/joeyespo/pytest-watch Is the "official" way to reexecute pytest --testmon on file changes.
